src/main/scala/chisel3/util/Decoupled.scala

```diff
@@ -227,20 +232,34 @@ class Queue[T <: Data](val gen: T,
   val do_deq = WireDefault(io.deq.fire())
 
   when (do_enq) {
-    ram(enq_ptr.value) := io.enq.bits
+    if (useSyncReadMem) {
```

Comment:
Is this elaboration conditional actually necessary? Both cases should do the same thing.

Reply:
No, obviously not necessary, but vast majority of the code that deals with `SyncReadMem` uses `read` and `write` methods for appropriate read and write processes. Right now this `if`/`else` construct is eliminated, as it was in the original code, but it can be easily reverted back if it turns out to be good to have it.